src/ydata/sdk/synthesizers/synthesizer.py

```diff
@@ -386,8 +386,16 @@ def _resolve_api_status(api_status: Dict) -> Status:
             Synthesizer Status
         """
         status = Status(api_status.get('state', Status.UNKNOWN.name))
-        prepare = PrepareState(api_status.get('prepare', {}).get(
-            'state', PrepareState.UNKNOWN.name))
-        if prepare == PrepareState.FAILED:
-            status = Status.FAILED
+        if status == Status.PREPARE:
+            if PrepareState(api_status.get('prepare', {}).get(
+                    'state', PrepareState.UNKNOWN.name)) == PrepareState.FAILED:
+                return Status.FAILED
+        elif status == Status.TRAIN:
+            if TrainingState(api_status.get('training', {}).get(
+                    'state', TrainingState.UNKNOWN.name)) == TrainingState.FAILED:
+                return Status.FAILED
+        elif status == Status.REPORT:
+            if ReportState(api_status.get('report', {}).get(
+                    'state', ReportState.UNKNOWN.name)) == ReportState.FAILED:
+                return Status.FAILED
```

Comment:
Seems a bit convoluted to me, because I don't know why we have multiple status/state enums, but without removing them there any solution will be unnecessarily complicated

Reply:
The multiple statuses/states are related to the state machine used in the backend. There is no way around that.